cli/dcoscli/log.py

```diff
@@ -0,0 +1,199 @@
+import functools
+import sys
+import time
+
+from dcos import emitting, util
+from dcos.errors import DCOSException
+
+logger = util.get_logger(__name__)
+emitter = emitting.FlatEmitter()
+
+
+def _no_file_exception():
+    return DCOSException('No files exist. Exiting.')
+
+
+def log_files(mesos_files, follow, lines):
+    """Print the contents of the given `mesos_files`.  Behaves like unix
+    tail.
+
+    :param mesos_files: file objects to print
+    :type mesos_files: [MesosFile]
+    :param follow: same as unix tail's -f
+    :type follow: bool
+    :param lines: number of lines to print
+    :type lines: int
+    :rtype: None
+    """
+
+    fn = functools.partial(_read_last_lines, lines)
+    curr_header, mesos_files = _stream_files(None, fn, mesos_files)
+    if not mesos_files:
+        raise _no_file_exception()
+
+    while follow:
+        # This flush is needed only for testing, since stdout is fully
+        # buffered (as opposed to line-buffered) when redirected to a
+        # pipe.  So if we don't flush, our --follow tests, which use a
+        # pipe, never see the data
+        sys.stdout.flush()
```

Comment:
:+1:. I think we have a similar problem when we prompt the user if they want to install the package.